### PR TITLE
Thread messages and runs

### PR DESCRIPTION
**Pull request by khoroshevj**
I'm finishing run tests here and will be glad to hear feedback before processing further.
Also can split into two PRs "messages" and "runs"

### CODE REVIEW

**Comment by StephenHodgson** [comment]
Lots of conflicts. I assumed you were done, so I started cleaning up what you gave me.

File: OpenAI-DotNet-Tests/OpenAI-DotNet-Tests.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (StephenHodgson) [inline OpenAI-DotNet-Tests/OpenAI-DotNet-Tests.csproj:3]: revert

Comment (khoroshevj) [reply OpenAI-DotNet-Tests/OpenAI-DotNet-Tests.csproj:3]: sure, just for local run

**Comment by khoroshevj** [comment]
> Lots of conflicts. I assumed you were done, so I started cleaning up what you gave me.

thats not a bit problem I guess, ping me when you finish I'll merge/rebase

**Comment by khoroshevj** [comment]
@StephenHodgson I'm done here and will be ready to rebase when you merge https://github.com/RageAgainstThePixel/OpenAI-DotNet/pull/168 into the main

**Comment by StephenHodgson** [comment]
Go ahead and rebase and get this in, and I'll finish my work later

**Comment by khoroshevj** [comment]
created new PR, closing this one

**Pull request closed without merging**